internal/semver/semver.go

```diff
@@ -70,13 +138,24 @@ func (v Semver) Compare(other Semver) int {
 
 // MajorMinorEqual returns if the major and minor version of two versions are equal.
 func (v Semver) MajorMinorEqual(other Semver) bool {
-	return v.Major == other.Major && v.Minor == other.Minor
+	return v.major == other.major && v.minor == other.minor
 }
 
 // IsUpgradeTo returns if a version is an upgrade to another version.
 // It checks if the version of v is greater than the version of other and allows a drift of at most one minor version.
-func (v Semver) IsUpgradeTo(other Semver) bool {
-	return v.Compare(other) > 0 && v.Major == other.Major && v.Minor-other.Minor <= 1
+func (v Semver) IsUpgradeTo(other Semver) error {
```

Comment:
I think that this logic should not be part of the type, since we support different upgrades for the different version types (microService, node image).
Instead, i'd have a function 
`compatibility.IsMicroServiceUpgrade()` , the alternative of having seperate types for each version type seems overkill